sql/catalog: TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces appears to be flaky

See https://teamcity.cockroachdb.com/viewLog.html?buildId=2149254&tab=buildResultsDiv&buildTypeId=Cockroach_EssentialCi on a branch for a seemingly unrelated change.

```
panic: test timed out after 30m0s

goroutine 70211 [running]:
testing.(*M).startAlarm.func1()
        /usr/local/go/src/testing/testing.go:1459 +0xdf
created by time.goFunc
        /usr/local/go/src/time/sleep.go:168 +0x44

goroutine 1 [chan receive, 29 minutes]:
testing.(*T).Run(0xc003462b40, 0x444fb80, 0x33, 0x4542678, 0x73ec01)
        /usr/local/go/src/testing/testing.go:1043 +0x37e
testing.runTests.func1(0xc00020fb00)
        /usr/local/go/src/testing/testing.go:1284 +0x78
testing.tRunner(0xc00020fb00, 0xc00082bdf0)
        /usr/local/go/src/testing/testing.go:991 +0xdc
testing.runTests(0xc00087eee0, 0x729f620, 0x25, 0x25, 0x0)
        /usr/local/go/src/testing/testing.go:1282 +0x2a7
testing.(*M).Run(0xc0004cca00, 0x0)
        /usr/local/go/src/testing/testing.go:1199 +0x15f
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease_test.TestMain(0xc0004cca00)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/lease/main_test.go:32 +0xb6
main.main()
        _testmain.go:118 +0x135

...

goroutine 41406 [chan receive, 29 minutes]:
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease_test.TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces(0xc003462b40)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/lease/lease_test.go:2356 +0x6f5
testing.tRunner(0xc003462b40, 0x4542678)
        /usr/local/go/src/testing/testing.go:991 +0xdc
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1042 +0x357
```

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -2337,7 +2337,7 @@ func TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces(t *testing.T) {
 	interestingTable.Store(tableID)
 
 	// Launch a goroutine to query foo. It will be blocked in lease acquisition.
-	selectDone := make(chan error)
+	selectDone := make(chan error, 1)
 	go func() {
 		var count int
 		selectDone <- db2.QueryRow("SELECT count(*) FROM foo").Scan(&count)
@@ -2348,7 +2348,7 @@ func TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces(t *testing.T) {
 
 	// Launch a goroutine to perform a schema change which will lead to new
 	// versions.
-	alterErrCh := make(chan error)
+	alterErrCh := make(chan error, 1)
 	go func() {
 		_, err := db1.Exec("ALTER TABLE foo ADD COLUMN j INT DEFAULT 1")
 		alterErrCh <- err
@@ -2356,8 +2356,14 @@ func TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces(t *testing.T) {
 
 	// Make sure we get an update. Note that this is after we have already
 	// acquired a lease on the old version but have not yet recorded that fact.
-	desc := <-descUpdateChan
-	require.Equal(t, descpb.DescriptorVersion(2), sqlbase.GetDescriptorVersion(desc))
+	select {
+	case err := <-alterErrCh:
+		t.Fatalf("alter succeeded before expected: %v", err)
+	case err := <-selectDone:
+		t.Fatalf("select succeeded before expected: %v", err)
+	case desc := <-descUpdateChan:
+		require.Equal(t, descpb.DescriptorVersion(2), sqlbase.GetDescriptorVersion(desc))
+	}
 
 	// Allow the original lease acquisition to proceed.
 	close(toUnblockForLeaseAcquisition)

